src/systemcmds/param/param.cpp

```diff
@@ -783,7 +821,62 @@ do_set(const char *name, const char *val, bool fail_on_not_found)
 }
 
 static int
-do_compare(const char *name, char *vals[], unsigned comparisons, enum COMPARE_OPERATOR cmp_op, enum COMPARE_ERROR_LEVEL err_level)
+do_set_custom_default(const char *name, const char *val)
+{
+	param_t param = param_find(name);
+
+	/* set nothing if parameter cannot be found */
+	if (param == PARAM_INVALID) {
+		/* param not found - fail silenty in scripts as it prevents booting */
+		PX4_ERR("Parameter %s not found.", name);
+		return PX4_ERROR;
+	}
+
+	// Set parameter if type is known and conversion from string to value turns out fine
+	switch (param_type(param)) {
+	case PARAM_TYPE_INT32: {
+			int32_t i;
+
+			if (param_get_default_value(param, &i) == PX4_OK) {
+				/* convert string */
+				char *end;
+				int32_t newval = strtol(val, &end, 10);
+
+				if ((i != newval) && (param_set_default_value(param, &newval) == PX4_OK)) {
+					PARAM_PRINT(" parameter default: %s %d -> %d\n", param_name(param), i, newval);
```

Comment:
Do we get these now on each bootup?

Reply:
I added it initially for visibility, but with things now working I can get rid of it and add some other mechanism to directly query the custom defaults set.